packages/wrangler/src/delete.ts

```diff
@@ -98,3 +115,130 @@ async function deleteSiteNamespaceIfExisting(
 		logger.log(`🌀 Deleted asset namespace for Workers Site "${ns.title}"`);
 	}
 }
+
+type ServiceReference = {
+	name: string;
+	service: string;
+	environment: string;
+};
+
+type DurableObjectServiceReference = ServiceReference & {
+	durable_object_namespace_id: string;
+	durable_object_namespace_name: string;
+};
+
+type DispatchOutboundsServiceReference = ServiceReference & {
+	namespace: string;
+	params: { name: string }[];
+};
+
+export type ServiceReferenceResponse = {
+	services?: {
+		incoming: ServiceReference[];
+		outgoing: ServiceReference[];
+	};
+	durable_objects?: DurableObjectServiceReference[];
+	dispatch_outbounds?: DispatchOutboundsServiceReference[];
+};
+
+export type Tail = {
+	tag: string;
+	producer:
+		| {
+				service: string;
+				environment?: string;
+		  }
+		| { script: string };
+	consumer:
+		| {
+				service: string;
+				environment?: string;
+		  }
+		| { script: string };
+	log_push?: boolean;
+	config?: unknown;
+	created_on: string;
+	modified_on: string;
+};
+
+type ExtraConfirmResult = "not_needed" | "needed_denied" | "needed_confirmed";
+
+type ScriptDetails =
+	| {
+			service: string;
+			environment?: string;
+	  }
+	| {
+			script: string;
+	  };
+
+function renderScriptName<T extends ScriptDetails = ScriptDetails>(details: T) {
+	let service: string;
+	let environment: string | undefined;
+	if ("script" in details) {
+		service = details.script;
+	} else {
+		service = details.service;
+		environment = details.environment;
+	}
+	return environment ? `${service} (${environment})` : service;
+}
+
+async function checkAndConfirmForceDeleteIfNecessary(
+	scriptName: string,
+	accountId: string
+): Promise<ExtraConfirmResult> {
+	const references = await fetchResult<ServiceReferenceResponse>(
+		`/accounts/${accountId}/workers/scripts/${scriptName}/references`
+	);
+	const tailProducers = await fetchResult<Tail[]>(
+		`/accounts/${accountId}/workers/tails/by-consumer/${scriptName}`
+	);
+	const isDependentService =
+		(references.services?.incoming.length || 0) > 0 ||
+		(references.durable_objects?.filter((ref) => ref.service !== scriptName)
+			?.length || 0) > 0 ||
+		(references.dispatch_outbounds?.length || 0) > 0 ||
+		tailProducers.length > 0;
```

Comment:
This is a fairly complex conditional—could it be refactored into multiple variables/lifted out into a separate function?

Reply:
refacored out into helper fns